pkg/controllers/rollout/controller_test.go

```diff
@@ -1503,141 +1451,787 @@ func TestPickBindingsToRoll(t *testing.T) {
+					TargetCluster:        cluster4,
+					ResourceSnapshotName: "snapshot-1",
 				},
 			},
+			wantTobeUpdatedBindings:     []int{2}, // specified MaxSurge helps us pick one scheduled binding to rollout. we don't have any ready binding so we don't remove any binding.
+			wantStaleUnselectedBindings: []int{3},
+			wantNeedRoll:                true,
+			wantWaitTime:                time.Second,
 		},
-	}
-}
-
-func clusterResourcePlacementForTest(crpName string, policy *fleetv1beta1.PlacementPolicy) *fleetv1beta1.ClusterResourcePlacement {
-	return &fleetv1beta1.ClusterResourcePlacement{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: crpName,
-		},
-		Spec: fleetv1beta1.ClusterResourcePlacementSpec{
-			ResourceSelectors: []fleetv1beta1.ClusterResourceSelector{
-				{
-					Group:   corev1.GroupName,
-					Version: "v1",
-					Kind:    "Namespace",
-					LabelSelector: &metav1.LabelSelector{
-						MatchLabels: map[string]string{"region": "east"},
+		"test policy change with MaxUnavailable specified, evict resources on unscheduled cluster - rollout allowed for one unscheduled binding": {
+			allBindings: []*fleetv1beta1.ClusterResourceBinding{
+				setDeletionTimeStampForBinding(generateReadyClusterResourceBinding(fleetv1beta1.BindingStateUnscheduled, "snapshot-1", cluster1)),
+				generateReadyClusterResourceBinding(fleetv1beta1.BindingStateUnscheduled, "snapshot-1", cluster2),
+				generateClusterResourceBinding(fleetv1beta1.BindingStateScheduled, "snapshot-1", cluster3),
+				generateClusterResourceBinding(fleetv1beta1.BindingStateScheduled, "snapshot-1", cluster4),
+			},
+			latestResourceSnapshotName: "snapshot-1",
+			crp: clusterResourcePlacementForTest("test",
+				createPlacementPolicyForTest(fleetv1beta1.PickNPlacementType, 2),
+				createPlacementRolloutStrategyForTest(fleetv1beta1.RollingUpdateRolloutStrategyType, &fleetv1beta1.RollingUpdateConfig{
+					MaxUnavailable: &intstr.IntOrString{
+						Type:   intstr.Int,
+						IntVal: 2,
 					},
+					MaxSurge: &intstr.IntOrString{
+						Type:   intstr.Int,
+						IntVal: 0,
+					},
+					UnavailablePeriodSeconds: ptr.To(1),
+				}, nil)),
+			wantDesiredBindingsSpec: []fleetv1beta1.ResourceBindingSpec{
+				{},
+				{},
+				{
+					State:                fleetv1beta1.BindingStateBound,
+					TargetCluster:        cluster3,
+					ResourceSnapshotName: "snapshot-1",
+				},
+				{
+					State:                fleetv1beta1.BindingStateBound,
+					TargetCluster:        cluster4,
+					ResourceSnapshotName: "snapshot-1",
 				},
 			},
-			Policy: policy,
-			Strategy: fleetv1beta1.RolloutStrategy{
-				Type: fleetv1beta1.RollingUpdateRolloutStrategyType,
-				RollingUpdate: &fleetv1beta1.RollingUpdateConfig{
+			wantTobeUpdatedBindings:     []int{1}, // specified MaxUnavailable helps us remove unscheduled binding, deleting unscheduled binding even though ready is a canBeUnavailable binding.
+			wantStaleUnselectedBindings: []int{2, 3},
```

Comment:
add a comment here too?  something like we can't pick them because the max Surge is 0?